### PR TITLE
Fix OIDC redirect URI computation

### PR DESCRIPTION
**Pull request by david-oc-miller**

### CODE REVIEW

File: src/app/charts/core/templates/_helpers-sso.tpl [hunk 1]
```diff
@@ -188,15 +190,10 @@ idp.xml
     {{- fail (printf "The OIDC configuration must be given as a map, not a %s: %s" (kindOf $oidc) $oidc) -}}
   {{- end -}}
 
-  {{- /* Set the usersDirectory for all clients to the default one ... */ -}}
-  {{- $baseUrl := (include "arkcase.tools.conf" (dict "ctx" $ctx "value" "baseUrl")) -}}
```
Comment (drivera-armedia) [inline src/app/charts/core/templates/_helpers-sso.tpl:192]: See this ... referenced below

**Review by drivera-armedia — CHANGES_REQUESTED** [review]

File: src/app/charts/core/templates/_helpers-sso.tpl (hunk 1 above)
Comment (david-oc-miller) [reply src/app/charts/core/templates/_helpers-sso.tpl:192]: Interesting ... with the code in the new location the lookup works.  The original code always returned  the default value from src/app/charts/core/values.yaml.

**Comment by david-oc-miller** [comment]
fixes submitted

File: src/app/charts/core/templates/_helpers-sso.tpl [hunk 2]
```diff
@@ -164,12 +164,13 @@ idp.xml
...
 
-      {{- /* Set the usersDirectory to the specified value */ -}}
-      {{- $client = set $client "usersDirectory" $usersDirectory -}}
-
       {{- /* Set the redirectUri to the specified value */ -}}
       {{- $client = set $client "redirectUri" (printf "%s/%s" $baseUrl $client.registrationId) -}}
 
+      {{- $usersDirectory := ((hasKey $client "usersDirectory") | ternary $client.usersDirectory $defaultUsersDirectory) -}}
+      {{- $client = set $client "usersDirectory" $usersDirectory -}}
+
```
Comment (drivera-armedia) [inline src/app/charts/core/templates/_helpers-sso.tpl:172]: Please don't add empty lines with leading spaces :D

If we're going to demand an end to sloppiness, it needs to start with us (to the best of our abilities)

File: src/app/charts/core/templates/_helpers-sso.tpl [hunk 3]
```diff
@@ -182,21 +183,20 @@ idp.xml
   {{- if not (include "arkcase.isRootContext" $ctx) -}}
     {{- fail "The 'ctx' parameter must be the root context ($ or .)" -}}
   {{- end -}}
+
```
Comment (drivera-armedia) [inline src/app/charts/core/templates/_helpers-sso.tpl:186]: Please remove the spaces on this line

File: src/app/charts/core/templates/_helpers-sso.tpl (hunk 1 above)
Comment (drivera-armedia) [reply src/app/charts/core/templates/_helpers-sso.tpl:192]: That's odd indeed ... the `arkcase.tools.conf` template was meant to facilitate looking configuration values, specifically, no matter how/where they'd be housed. This makes it easy to fetch them without knowing their full whereabouts, while also giving the maintainers freedom to restructure the configuration if desired.

I'll have to look into why one worked but not the other ...

File: src/app/charts/core/templates/_helpers-sso.tpl [hunk 4]
```diff
@@ -267,8 +267,10 @@ idp.xml
         {{- $protocol = ($saml | ternary "saml" "oidc") -}}
       {{- end -}}
 
+      {{- $baseUrl := (include "arkcase.tools.conf" (dict "ctx" $ctx "value" "baseUrl")) -}}
+
```
Comment (drivera-armedia) [inline src/app/charts/core/templates/_helpers-sso.tpl:271]: Remember to remove the trailing slashes:

`$baseUrl = (regexReplaceAll "/*$" $baseUrl "")`

**Review by drivera-armedia — COMMENTED** [review]
On further review, these changes are unnecessary with one exception: they did uncover a bug which was causing the wrong `baseUrl` value to be used when rendering the OIDC configurations. This has been resolved in 0.8.0-42.

- The `usersDirectory` value must always be `arkcase` because that's the only place where SSO makes sense (right?), and in these new charts there are only two valid directory names: `arkcase` and `portal` ... gone are the days of having a directory named after a domain. The upgrade bootup code cleans this up in the DB as well ;)
- Since the `baseUrl` value is now being rendered correctly, there's no further need for modifying the existing code to correct this.

**Comment by drivera-armedia** [comment]
As described above, upon further review, these changes aren't necessary.

**Pull request closed without merging**